File: crates/deslop/tests/fixtures/fused-golden-typescript/shape_beta.ts
export class BillingDescriptor {
  readonly region = "beta-billing";
  readonly attempts = 9;
  readonly origin = "https://beta.example.com/billing";

  assemble(account: string): string {
    return this.origin + "/" + account + "/" + this.region + "?attempts=" + String(this.attempts);
  }
}

File: crates/deslop/tests/fixtures/fused-golden-rust/shape_beta.rs
pub struct BillingDescriptor {
    pub region: String,
    pub attempts: i64,
    pub origin: String,
}

impl BillingDescriptor {
    pub fn assemble(&self, account: &str) -> String {
        format!("{}/{}/{}?attempts=beta", self.origin, account, self.region)
    }
}
